patches/2025-10-31-patch-device-primary-names-and-balances.lua

```diff
@@ -21,41 +32,36 @@ local function _loaded_mod_src_hb()
 	-- hb.lua needs to be in its own file and not in balances.lua to avoid circular dependencies
 	local hb = {}
 
-	---@param oldBalances table<string, number> A table of addresses and their balances
-	---@return table<string, boolean> affectedBalancesAddresses table of addresses that have had balance changes
-	function hb.patchBalances(oldBalances)
-		assert(type(oldBalances) == "table", "Old balances must be a table")
+	---@return table<string, string>|nil affectedBalancesAddresses table of addresses and their balance values as strings
+	function hb.createBalancesPatch()
 		local affectedBalancesAddresses = {}
-		for address, _ in pairs(oldBalances) do
-			if Balances[address] ~= oldBalances[address] then
+		for address, _ in pairs(Balances) do
+			if HyperbeamSync.balances[address] ~= Balances[address] then
 				affectedBalancesAddresses[address] = true
 			end
 		end
-		for address, _ in pairs(Balances) do
-			if oldBalances[address] ~= Balances[address] then
+
+		for address, _ in pairs(HyperbeamSync.balances) do
+			if Balances[address] ~= HyperbeamSync.balances[address] then
 				affectedBalancesAddresses[address] = true
 			end
 		end
 
 		--- For simplicity we always include the protocol balance in the patch message
-		--- this also prevents us from sending an empty patch message and deleting the entire hyperbeam balances table\
+		--- this also prevents us from sending an empty patch message and deleting the entire hyperbeam balances table
```

Comment:
I think we have enough experience with the system now to stop doing this. Let's just make sure we never send an empty balances list unless there's truly no balances.

Reply:
[4ce76ec](https://github.com/ar-io/ar-io-network-process/pull/442/commits/4ce76ec31c824ad75d8380bde90f0e510e2526eb)